server/mergin/sync/models.py

```diff
@@ -833,7 +834,71 @@ def files(self) -> List[ProjectFile]:
             WHERE fh.change != 'delete';
         """
         params = {"project_id": self.project_id, "version": self.name}
-        result = db.session.execute(query, params).fetchall()
+        return db.session.execute(query, params).fetchall()
+
+    def _files_backward_search(self):
+        """Calculate version files using lookup from the last version"""
+        query = f"""
+            WITH files_changes_before_version AS (
+                WITH files_candidates AS (
+                    -- files removed later (but created anytime)
+                    SELECT DISTINCT
+                        fh.file_path_id AS file_id
+                    FROM project_version pv
+                    LEFT OUTER JOIN file_history fh ON fh.version_id = pv.id
+                    WHERE
+                        pv.project_id = :project_id
+                        AND pv.name > :version
+                        AND fh.change = 'delete'
+                    -- union with current files
+                    UNION
```

Comment:
Just question: Are you sure with ```UNION``` or you want to merge It with ```UNION ALL``` without any deduplication?


    UNION: only keeps unique records
    UNION ALL: keeps all records, including duplicates